documentation/decisions/adr/2023-12-18-measurement-dashboard-architecture.md

```diff
@@ -0,0 +1,149 @@
+# Public measurement dashboard architecture
+
+- **Status:** Active <!-- REQUIRED -->
+- **Last Modified:** 2023-12-18 <!-- REQUIRED -->
+- **Related Issue:** [#845](https://github.com/HHS/simpler-grants-gov/issues/845) <!-- RECOMMENDED -->
+- **Deciders:**
+  - Lucas Brown
+  - Aaron Couch
+  - Michael Chouinard
+  - Billy Daly
+- **Tags:** analytics, data <!-- OPTIONAL -->
+
+## Context and Problem Statement
+
+One of the goals for this project is to make all of the Simpler.Grants.gov operational metrics publicly available and to publish as much non-sensitive data about the grants pipeline as possible. In order to help internal and external stakeholders make sense of this data, we'll need to create and maintain a set of public measurement dashboards that summarize and contextualize the key metrics we track in the project.
+
+There are a number of architectural patterns we can follow to build and publish a set of dashboards, and the pattern we choose will influence how we set up our data engineering pipeline, as well as our infrastructure for analytics and business intelligence (ABI).
+
+The goal of this ADR is to evaluate and recommend an architectural approach for our initial public measurement dashboard deliverable, which will focus on sprint and delivery metrics. However, this recommendation should also create a flexible yet robust foundation for future dashboards as well.
+
+## Decision Drivers <!-- RECOMMENDED -->
+
+#### Must have
+
+- Members of the public can access the dashboard without a login
+- Multiple related dashboards can be combined into a single "application" for external users
+- New charts and dashboards are easy to implement
+- Users can export or download the data behind a given dashboard
+- Open source contributors can host a local version of our dashboards using exported data
+
+#### Nice to have
+
+- The platform used to build dashboards can also be used to answer ad-hoc questions about the data
+- System-to-system users can programmatically access the data behind a given dashboard via API
+- Dashboards can include narrative text that explains the metrics we publish
+
+## Options Considered
```

Comment:
Maybe add an option: some combination of the above? 

On one private sector analytics project I worked on, we would: 

1. Develop analytics using raw SQL and Excel, and if these one-off reports were useful to key stakeholders, we would then...
2. Develop dashboards in Tableau that got updated nightly, and if these were useful to stakeholders, we would then...
3. Build them into our Ruby-on-Rails app. 

Something similar here could be: 
1. Prototype in Jupyter, and if it's highly useful then we...
2. Build into a dashboard, and if it's highly used then we...
3. Build it into an API. 

I'm not sure that's actually what we want to do, but it's an option to consider.

Reply:
An alternative way to combine these would be: 

1. Run an ETL script that takes data from the production DB and aggregates it, removing all PII or small-sample data. This data could be made available over an API, either an advanced API like a RESTful JSON API, or a simple API such as CSVs posted to S3 in a predictable format. 
2. We put either custom dashboards, or open source / SaaS dashboards, that point to the aggregated data as a source of truth. 

This two-step process enables us to make the data from 1 publicly accessible, while still giving us flexibility on the business intelligence tools for step 2. We'd no longer need to worry about exposing APIs in step 2 because the data was already exposed in step 1.

Reply:
I added an option for combining s3 bucket + dashboard UI (e.g. jupyter notebook, static site, etc.) and then recommended we use this approach in the short-term and then in the long-term combine open source dashboard tool and an API + dashboard UI in the iterative workflow you describe above:

- [feat: Adds s3 bucket + user interface option](https://github.com/HHS/simpler-grants-gov/pull/936/commits/ca39b6783edbe3eef557d69f197b87158fc1a046)
- [feat: Adds recommendation for dashboard architecture](https://github.com/HHS/simpler-grants-gov/pull/936/commits/201ee17416369478f43b6590bb12a0c9cf878bf0)